src/unix/thread.c

```diff
@@ -72,10 +75,59 @@ int uv_thread_create(uv_thread_t *tid, void (*entry)(void *arg), void *arg) {
 }
 
 
+int uv_thread_create_on(int proc_num,
+                        uv_thread_t *tid,
+                        void (*entry)(void *arg),
+                        void *arg) {
```

Comment:
I'm not wildly in love with the API.  It's a one-off for a niche feature and not very flexible at that.  A versioned attribute struct would help address that:

```c
int uv_thread_create_with_attr(uv_thread_t* tid, uv_thread_attr_t* attr, void (*entry)(void*), void* arg));
```

Where `uv_thread_attr_t` would look like this:

```c
typedef struct uv_thread_attr_s uv_thread_attr_t;

enum uv_thread_attr_version {
  UV_THREAD_ATTR_VERSION_1
  /* Macro for feature detection. */
#define UV_THREAD_ATTR_VERSION_1 UV_THREAD_ATTR_VERSION_1
};

struct uv_thread_attr_v1 {
  char* cpumask;
  size_t cpumask_size;  /* TBD if expressed in bits or bytes. */
};

struct uv_thread_attr_s {
  int version;
  union {
    struct uv_thread_attr_v1 v1;
  } data;
};
```

That way we can always add v2, v3, etc. versions with new features later on.